SIMSBIOHUB-424: Publish Documents to BioHub from SIMS

KjartanE opened the pull request:
## Links to Jira Tickets
- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-424

## Description of Changes
- Send artifacts to BioHub as part of the existing publish workflow.
- The artifacts are to be sent as separate requests 
- When a survey has attached artifacts, those artifacts are submitted to biohub during publish

## Testing Notes

- {List any relevant testing considerations, necessary pre-reqs, and areas of the app to focus on. Specifically, include anything that will help the reviewers of this PR verify the code is functioning as expected.}

Related PR for the BioHub changes: https://github.com/bcgov/biohubbc-platform/pull/230

KjartanE committed:
added initial attachment submission

KjartanE committed:
cleanup, platform artifact submission setup

KjartanE committed:
swap name to submission_uuid

KjartanE committed:
wip, attachment artifact submission

KjartanE committed:
SIMSBIOHUB-395: Dev - Bulk Delete Sampling Sites (#1191)

- Select one or many records
- Bulk action to delete those records.
- Prevent deletion of records that are referenced by other data (e.g. Sampling site is referenced by an observation).

KjartanE committed:
initial attachment submission accepted

KjartanE committed:
history publish table clean up

KjartanE committed:
SIMSBIOHUB-395-2: Select all bulk delete sample sites (#1192)

* bulk delete sample sites, select all
* pr edit, add indeterminate on checkbox

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-432

KjartanE committed:
fix broken tests

curtisupshall commented on api/src/paths/project/{projectId}/survey/{surveyId}/view.ts:
```diff
@@ -399,9 +399,8 @@ GET.apiDoc = {
-                        type: 'integer',
-                        minimum: 1
+                      submission_uuid: {
+                        type: 'string'
```
Nit: add `format: 'uuid'`

curtisupshall commented on api/src/paths/project/{projectId}/view.ts:
```diff
@@ -213,9 +213,8 @@ GET.apiDoc = {
-                        type: 'integer',
-                        minimum: 1
+                      submission_uuid: {
+                        type: 'string'
```
Nit: add `format: 'uuid'`

curtisupshall commented on api/src/paths/publish/survey.ts:
```diff
@@ -71,8 +71,9 @@ POST.apiDoc = {
-                type: 'number'
+              submission_uuid: {
+                type: 'string',
+                description: 'The UUID of the submission'
```
Nit: add `format: 'uuid'`

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-432

KjartanE committed:
fix s3_key prop

curtisupshall commented on api/src/repositories/history-publish-repository.ts:
```diff
@@ -6,49 +6,49 @@ import { BaseRepository } from './base-repository';
   project_id: z.number(),
   event_timestamp: z.string(),
-  queue_id: z.number(),
+  submission_uuid: z.string(),
```
Can use `z.string().uuid()`

curtisupshall committed:
SIMSBIOHUB-432: Added 'uuid' format constraint to openAPI and Zod schemas

curtisupshall reviewed (dismissed):
LGTM

al-rosenthal committed:
Merge branch 'dev' into SIMSBIOHUB-432

KjartanE committed:
Merge branch 'dev' into SIMSBIOHUB-432

NickPhura committed:
Update to match latest BioHub submission endpoint request requirements

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-432

NickPhura committed:
Tweak artifact submit properties

NickPhura committed:
Merge branch 'SIMSBIOHUB-432' of https://github.com/bcgov/biohubbc into SIMSBIOHUB-432

NickPhura committed:
Minor simplification to platform artifact submission.
Remove delete artifact from biohub code (was causing regular sims artifact deletion to fail)

NickPhura committed:
Fix tests.
Add support for submitting description + comment to biohub.

curtisupshall reviewed (dismissed):
Static read of the code looks good to me 📑

NickPhura committed:
Add support for submitting reports to BioHub

NickPhura committed:
Tweak artifact submission/tests.
ignore-skip

NickPhura committed:
Formatting.

curtisupshall reviewed (approved):
LGTM

al-rosenthal reviewed (approved):
ran this locally and it works as expected! 🌮
